apps/ewallet_db/lib/ewallet_db/state_machines/transaction_state.ex

```diff
@@ -32,9 +34,11 @@ defmodule EWalletDB.TransactionState do
     @pending => {[], []},
     @confirmed => {[:local_ledger_uuid], []},
     @failed => {[:error_code, :error_description, :error_data], [:error_code]},
+    @blockchain_failed => {[:error_code, :error_description, :error_data], [:error_code]},
     @blockchain_submitted => {[:blockchain_tx_hash], [:blockchain_tx_hash]},
     @ledger_pending => {[], []},
     @pending_confirmations => {[:confirmations_count], [:confirmations_count]},
+    @ledger_blockchain_confirmed => {[:confirmations_count], [:confirmations_count]},
```

Comment:
Can you document as comment the differences between `@ledger_pending`, `@pending_confirmations`, `@ledger_blockchain_confirmed`, `@blockchain_confirmed`, `@ledger_confirmed`?